src/Stack/Types/GhcPkgId.hs

```diff
@@ -36,13 +34,24 @@ instance Show GhcPkgIdParseFail where
 instance Exception GhcPkgIdParseFail
 
 -- | A ghc-pkg package identifier.
-data GhcPkgId =
-  GhcPkgId !PackageIdentifier
-           !ByteString
+newtype GhcPkgId = GhcPkgId ByteString
   deriving (Eq,Ord,Data,Typeable,Generic)
 
 instance Hashable GhcPkgId
-instance Binary GhcPkgId
+instance Binary GhcPkgId where
+    put (GhcPkgId x) = do
+        -- magic string
+        putWord8 1
+        putWord8 3
+        putWord8 4
+        putWord8 7
```

Comment:
Make sure that the binary format being read is the same one that was written. See: https://github.com/commercialhaskell/stack/issues/889